ops/testing.py

```diff
@@ -458,6 +485,63 @@ def add_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
         self._charm.on[relation_name].relation_joined.emit(
             relation, remote_unit.app, remote_unit)
 
+    def remove_relation_unit(self, relation_id: int, remote_unit_name: str) -> None:
+        """Remove a unit from a relation.
+
+        Example::
+
+          rel_id = harness.add_relation('db', 'postgresql')
+          harness.add_relation_unit(rel_id, 'postgresql/0')
+          ...
+          harness.remove_relation_unit(rel_id, 'postgresql/0')
+
+        This will trigger a `relation_departed` event. This would
+        normally be followed by a `relation_changed` event triggered
+        by Juju. However when using the test harness a
+        `relation_changed` event must be triggererd using
+        :meth:`.update_relation_data`. This deviation from normal Juju
+        behaviour, facilitates testing by making each step in the
+        charm life cycle explicit.
+
+        Args:
+            relation_id: The integer relation identifier (as returned by add_relation).
+            remote_unit_name: A string representing the remote unit that is being added.
+
+        Raises:
+            KeyError: if relation_id or remote_unit_name is not valid
+            ValueError: if remote_unit_name is not valid
+        """
+        relation_name = self._backend._relation_names[relation_id]
+
+        # gather data to invalidate cache later
+        if self._model is not None:  # defensive check for model
```

Comment:
But what happens if self._model is None? For sure you can not continue happily with the method, as for example 'unit_cache' will be not defined.

It looks a test is missing for this case, too.

Reply:
I am not sure under what conditions model can be none. Any ideas ? However if model is none then a relation unit will not be added in the first place. See for instance `add_relation_unit`, even though various data structures associated with the relation do get updated (c.f. `add_relation` and `add_relation_unit`). Hence in the process of removing relations the same logic is being inverted. I do not see what else could be done. Further if the `_model` is None then so will `unit_cache` and there is a defensive check for this when invalidating the unit cache.

Reply:
After investigating, here is what I find. It is not possible to add or remove a relation by forcing `harness.charm.framework.model` to be `None` because the `restore` method of `RelationEvent` access `model`.

Reply:
Removed check for model being none as per agreement in Standup.